src/dmd/mars.d

```diff
@@ -2383,3 +2470,47 @@ Modules createModules(ref Strings files, ref Strings libmodules)
     }
     return modules;
 }
+
+/**
+Prints a supplied usage text to the console and
+returns the exit code for the help usage page.
+
+Params:
+    help = usage text to be printed
+
+Returns:
+    `EXIT_SUCCESS` if no errors occurred, `EXIT_FAILURE` otherwise
+*/
+byte printHelpUsage(string help)
+{
+    printf("%.*s", help.length, &help[0]);
+    return global.errors ? EXIT_FAILURE : EXIT_SUCCESS;
+}
+
+
+/**
+Checks whether a flag has no options (e.g. -foo)
+
+Params:
+    flag = pointer to the end of the flag
+
+Returns: `true` if the flags has no options
+*/
+bool hasNoOption(const(char)* flag)
```

Comment:
This seems like quite a lot for `*flag == 0`.

Reply:
The function already got more complex as I realized that it should not only `-transition`, but also `-transition=` and trigger the same "ERROR: this switch needs arguments" message for both these cases.